src/OpenTK/DisplayDevice.cs

```diff
@@ -268,180 +212,58 @@ public static DisplayDevice GetDisplay(DisplayIndex index)
         }
 
         /// <summary>
-        /// Gets the original resolution of this instance.
+        /// Tries to find an available resolution for this DisplayDevice that matches the parameters passed in.
         /// </summary>
-        internal DisplayResolution OriginalResolution { get; set; }
-
-        internal static DisplayDevice FromPoint(int x, int y)
+        /// <param name="width"></param>
+        /// <param name="height"></param>
+        /// <param name="bitsPerPixel"></param>
+        /// <param name="refreshRate"></param>
+        /// <returns>A matching DisplayResolution or null if one cannnot be found.</returns>
+        DisplayResolution FindResolution(int width, int height, int bitsPerPixel, float refreshRate)
         {
-            for (DisplayIndex i = DisplayIndex.First; i < DisplayIndex.Sixth; i++)
+            foreach(var resolution in AvailableResolutions)
             {
-                DisplayDevice display = DisplayDevice.GetDisplay(i);
-                if (display != null)
+                if(
+                    ((width != 0 && width == resolution.Width) || width == 0) &&
+                    ((height != 0 && height == resolution.Height) || height == 0) &&
+                    ((bitsPerPixel != 0 && bitsPerPixel == resolution.BitsPerPixel) || bitsPerPixel == 0) &&
+                    ((refreshRate != 0 && System.Math.Abs(refreshRate - resolution.RefreshRate) < 1.0) || refreshRate == 0))
                 {
-                    if (display.Bounds.Contains(x, y))
-                    {
-                        return display;
-                    }
+                    return resolution;
                 }
             }
             return null;
```

Comment:
Leave a space after the closing brace. It sets what follows apart from the rest of the code.